Add new YSQL flags and update configuration options/flags for CLI section. (#2546)
yugabyte/yugabyte-db: docs/content/latest/admin/yb-tserver.md

--fs_data_dirs "/home/centos/disk1,/home/centos/disk2" &
```

## Online help

Run `yb-tserver --help` to display the online help.

```sh
$ ./bin/yb-tserver --help
```

## Syntax

```sh
yb-tserver [ options ]
```

## Configuration options

### General options

#### --help

Displays help on all options.

#### --helpon

Displays help on modules named by the specified option (or flag) value.

> #### --flagfile

Reviewer: We should move this below the mandatory options.

Author: Good catch.

Default: `never`

#### --webserver_interface

Address to bind for the web server user interface.

Default: `0.0.0.0`

#### --webserver_port

The port for monitoring the web server.

Default: `7000`

#### --webserver_doc_root

Monitoring web server home.

Default: The `www` directory in the YugabyteDB home directory.

### Logging options

#### --logtostderr

Flag to log to standard error (`stderr`).

#### --log_dir

Specifies the directory to store `yb-tserver` log files.

Default: Same as [`--fs_data_dirs`](#fs-data-dirs)

### Placement options

#### --placement_zone

Specifies the name of the availability zone, or rack, where this instance is deployed.

Default: `rack1`

#### --placement_region

Specifies the name of the region, or data center, where this instance is deployed.

Default: `datacenter1`

#### --placement_cloud

Specifies the name of the cloud where this instance is deployed.

Default: `cloud1`

### YSQL options

The following options, or flags, support the use of the [YSQL API](../../api/ysql/).

> #### --enable_ysql

Reviewer: Just a note: This will default to `true` soon in the 2.0.2 release.

Author: Got it -- can't wait :-)

Enables the YSQL API. Replaces the deprecated `--start_pgsql_proxy` option.

Default: `false`

#### --ysql_enable_auth

Enables YSQL authentication.

Default: `false`

#### --pgsql_proxy_bind_address

Specifies the bind address for the YSQL API.

Default: `0.0.0.0:5433`

#### --pgsql_proxy_webserver_port

Specifies the web server port for YSQL metrics monitoring.

Default: `13000`

#### --ysql_hba_conf

> Specify a comma-separated list of YugabyteDB setting assignments.

Reviewer: ```suggestion
Specifies a comma-separated list of PostgreSQL client authentication settings.
```

> Default: `""`

Reviewer: ```suggestion
Default: `"host all all 0.0.0.0/0 trust,host all all ::0/0 trust"`
```

#### --ysql_pg_conf

Comma-separated list of PostgreSQL setting assignments.

> Default: `""`

Reviewer: Should we just remove the Default line here and elsewhere below?

Author: I'm fine with that for now — a longer discussion later.

#### --ysql_timezone

Specifies the time zone for displaying and interpreting timestamps.

Default: `""` — Uses the YSQL time zone.

#### --ysql_datestyle

Specifies the display format for data and time values.

Default: `""` — Uses the YSQL display format.

#### --ysql_max_connections

Specifies the maximum number of concurrent YSQL connections.

> Default: `0`

Reviewer: ```suggestion
Default: `300`
```

#### --ysql_default_transaction_isolation

Specifies the default transaction isolation level.

> Default: `""`

Reviewer: ```suggestion
Default: `"repeatable read"`
```

Author: Changed to :

Default: `READ COMMITTED` (implemented in YugabyteDB as `REPEATABLE READ`)

Default: `false`

#### --cql_proxy_bind_address

Specifies the bind address for the YCQL API.

Default: `0.0.0.0:9042`

#### --cql_proxy_webserver_port

Specifies the port for monitoring YCQL metrics.

Default: `12000`

### YEDIS options

The following options, or flags, support the use of the YEDIS API.

#### --redis_proxy_bind_address

Specifies the bind address for the YEDIS API.

Default: `0.0.0.0:6379`

#### --redis_proxy_webserver_port

Specifies the port for monitoring YEDIS metrics.

Default: `11000`

### Performance options

#### --rocksdb_compact_flush_rate_limit_bytes_per_sec

Used to control rate of memstore flush and SSTable file compaction.

Default: `256MB`

#### --remote_bootstrap_rate_limit_bytes_per_sec

Rate control across all tablets being remote bootstrapped from or to this process.

Default: `256MB`

#### --yb_num_shards_per_tserver

The number of shards per YB-TServer per table when a user table is created.

> Default: `-1` — Server automatically picks a valid default internally.

Reviewer: ```suggestion
Default: Server automatically picks a valid default internally, typically 8.
```
